src/v/wasm/parser/tests/parser_test.cc

```diff
@@ -364,6 +367,91 @@ INSTANTIATE_TEST_SUITE_P(
   (memory 1)
   (data (i32.const 0) "a")
 )
+)WAT",
+        .exports = {},
+        .imports = {},
+      },
+      {
+        .wat = R"WAT(
+(module
+  (global (;0;) (mut i32) (i32.const 306464))
+  (global (;1;) i32 (i32.const 0))
+  (global (;2;) i32 (i32.const 239492))
+  (global (;3;) i32 (i32.const 448))
+  (global (;4;) i32 (i32.const 1))
+  (global (;5;) i32 (i32.const 193520))
+  (global (;6;) i32 (i32.const 449))
```

Comment:
This is the lucky guy who caused the bug :)

Keeping this full globals section (it's from a real Wasm file I tried to deploy) just for better coverage.